src/qml/NightstandWatchfacePage.qml

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2022 - Timo Könnecke <github.com/eLtMosen>
+ *               2022 - Darrel Griët <dgriet@gmail.com>
+ *               2015 - Florent Revest <revestflo@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.9
+import QtGraphicalEffects 1.12
+import Qt.labs.folderlistmodel 2.1
+import org.asteroid.controls 1.0
+import org.asteroid.utils 1.0
+import Nemo.Configuration 1.0
+import Nemo.Time 1.0
+
+Item {
+
+    property alias displayAmbient: compositor.displayAmbient
+    property string assetPath: "file:///usr/share/asteroid-launcher/"
+    property alias watchface: watchfaceNightstandSource.value
+
+    ConfigurationValue {
+        id: watchfaceNightstandSource
+        key: "/desktop/asteroid/nightstand/watchface"
+        defaultValue: assetPath + "watchfaces/005-analog-nordic.qml"
+    }
+
+    ConfigurationValue {
+        id: wallpaperSource
+        key: "/desktop/asteroid/background-filename"
+        defaultValue: "file:///usr/share/asteroid-launcher/wallpapers/full/000-flatmesh.qml"
+    }
+
+    ConfigurationValue {
+        id: use12H
+        key: "/org/asteroidos/settings/use-12h-format"
+        defaultValue: false
+    }
+
+    QtObject {
+        id: compositor
+        property bool displayAmbient: false
+    }
+
+    WatchfaceSelector {
```

Comment:
We are almost there I promise!

Right now WatchfaceSelector is a leaky abstraction because it expects a whole bunch of stuff to be defined in the context and this contract isn't clearly stated anywhere (the user needs to figure out that they are supposed to create all the properties above)

Actually, these are all (compositor, use12H, wallpaperSource, watchfaceSource, watchface, assetPath, and displayAmbient) duplicate code and deserve their spot in WatchfaceSelector too. If you move all of them into WatchfaceSelector and only expose a `property alias watchfaceSourceKey: watchfaceSource.key` you'll have abstracted everything away but the one thing which you actually care about and the API contract will be clearer for the user of WatchfaceSelector.

Reply:
Following off-line discussions, let's ignore this.

The existing codebase is already at a lower standard (eg: duplicate code between WallpaperPage and WatchfacePage) and it shouldn't be on the shoulders of contributors to fix the existing mess. This PR is already a strict improvement so should be merged. I already infringed enough rules of https://blog.gnoack.org/post/code_reviews/#behaviours-to-avoid-as-a-reviewer :) 

Mea culpa beroset!